### ISSUE TITLE
println("\u0007") renders as the charcter "a"

### ISSUE THREAD

**Issue by jfraudeau** [scala-js/scala-js#3125]
Trivial to reproduce, not sure how to implement as a test, `"\u0007" == "a"` still false

**Comment by jfraudeau** [comment]
Somehow `"a".replace("\u0007", "b")` is "b"

**Comment by sjrd** [comment]
`"\u0007" == "a"` is `false` because it is optimized away. The following shows `true`:
```scala
@noinline def id(s: String): String = s
println(id("\u0007") == "a")
```
So it has probably something to do with the pretty-printer to JS.

**Comment by jfraudeau** [comment]
```
def a(): String = {
    "\u0007"
  }
```
compiles to 
```
$c_L_ScalaJSTest.prototype.a__T = (function() {
  return "\a"
});
```

**Comment by sjrd** [comment]
OK lol. In JS, `\a` is not a valid escape sequence, so of course `"\a"` evaluates, without error, to `"a"` -_-'
Our pretty-printer in https://github.com/scala-js/scala-js/blob/d1b7cef76a3af33788ce24b1031d5e82b840a046/ir/src/main/scala/org/scalajs/core/ir/Utils.scala#L78 should therefore handle `7.toChar` like "normal" non-printable characters, and not like those that have a letter escape escape sequence (like `\n`).

**Comment by jfraudeau** [comment]
The fix will be easy to implement then 👍

**Comment by jducoeur** [comment]
Spent a little while scratching my head over this one, before discovering that `\a` *is* the correct escape character for this in Java.  Learn something new every day...

**Comment by jfraudeau** [comment]
@jducoeur It's not documented here : https://docs.oracle.com/javase/tutorial/java/data/characters.html
It seems to be the case with C (according to wikipedia... https://en.wikipedia.org/wiki/Escape_sequences_in_C)

**Comment by jducoeur** [comment]
@jfraudeau Interesting.  It *is* listed as an escape character for regular expressions, in [Pattern](https://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html)...

**Comment by sjrd** [comment]
Fixed in 5a9a54653e114bee76e55f3bc78d2c2d5dc3f34f.
